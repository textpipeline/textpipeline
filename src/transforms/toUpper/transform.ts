import { Transform } from '../types';

const transform: Transform = {
  name: 'To Uppercase',
  slug: 'to-upper',
  inputTypeName: 'Text',
  outputTypeName: 'Uppercased Text',
  openSourceProject: undefined,
  defaultOutput: '',
  execute: async (input: string) => input.toUpperCase(),
};

export default transform;
