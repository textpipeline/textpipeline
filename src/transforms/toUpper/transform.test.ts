import toUpper from './transform';

describe('toUpper', () => {
  it('has the correct attributes', () => {
    expect(toUpper).toMatchSnapshot();
  });

  describe('execute', () => {
    it('converts empty to empty', () => expect(toUpper.execute('')).resolves.toEqual(''));

    it('converts already upper cased text to itself', () =>
      expect(toUpper.execute('FOO BAR')).resolves.toEqual('FOO BAR'));

    it('converts mixed case to upper case', () => expect(toUpper.execute('Foo baR')).resolves.toEqual('FOO BAR'));

    it('converts lower case to upper case', () => expect(toUpper.execute('foo bar')).resolves.toEqual('FOO BAR'));

    it('converts text with non-alpha values correctly', () =>
      expect(toUpper.execute('foo bar 123 $%^&')).resolves.toEqual('FOO BAR 123 $%^&'));
  });
});
