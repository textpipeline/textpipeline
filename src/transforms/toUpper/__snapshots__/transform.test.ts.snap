// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toUpper has the correct attributes 1`] = `
Object {
  "defaultOutput": "",
  "execute": [Function],
  "inputTypeName": "Text",
  "name": "To Uppercase",
  "openSourceProject": undefined,
  "outputTypeName": "Uppercased Text",
  "slug": "to-upper",
}
`;
